src/auth/integration-tests/src/lib.rs

```diff
@@ -115,3 +120,118 @@ pub async fn api_key() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+pub async fn workload_identity_provider_url_sourced() -> anyhow::Result<()> {
+    let project = std::env::var("GOOGLE_CLOUD_PROJECT").expect("GOOGLE_CLOUD_PROJECT not set");
+    let audience = get_oidc_audience();
+    let service_account = get_byoid_service_account();
+    let service_account_data =
+        serde_json::from_value::<HashMap<String, String>>(service_account.clone())
+            .expect("failed to read service account data as map");
+    let client_email = service_account_data
+        .get("client_email")
+        .expect("missing client_email");
```

Comment:
This seems odd, are you trying to read a string from the JSON object?  Why not:

```suggestion
    let client_email = match service_account.get("client_email") {
      Some(serde_json::Value::String(v)) => v.clone(),
      None => panic!("missing `client_email` in data: {service_account:?}"),
      Some(_) => panic!("invalid service account config, `client_email` should be a string: {service_account:?}"),
    };
```

Reply:
good point, I forgot that you can read from a `Value` directly 🤦